lib/internal/Magento/Framework/Event/Invoker/InvokerDefault.php

```diff
@@ -61,23 +74,22 @@ public function dispatch(array $configuration, Observer $observer)
     }
 
     /**
+     * Execute Observer.
+     *
      * @param \Magento\Framework\Event\ObserverInterface $object
      * @param Observer $observer
      * @return $this
-     * @throws \LogicException
      */
     protected function _callObserverMethod($object, $observer)
     {
         if ($object instanceof \Magento\Framework\Event\ObserverInterface) {
             $object->execute($observer);
-        } elseif ($this->_appState->getMode() == \Magento\Framework\App\State::MODE_DEVELOPER) {
```

Comment:
I'm not sure that we should delete this condition for the developer mode, also such changes are backward incompatible because method signature is changed (LogicException is not throwing anymore).